### PR TITLE
diagnostic response

### PR DESCRIPTION
**Pull request by MFBrewster**

### CODE REVIEW

**Comment by MFBrewster** [comment]
Comfort: 4,
Clarity: 3

Used class notes and the node console

File: lib/diagnostic.js [hunk 1]
```diff
@@ -14,20 +14,21 @@
...
 
-// Which scope(s) may access the variable `laswerQuantity`? Choose one or many
+// Which scope(s) may access the variable `laserQuantity`? Choose one or many
 // of the following:
 //
 //     `['alderaan', 'deathStar', 'fire', 'global']`
 //
-let scopes = [/* your response here */];
+let scopes = ['deathStar', 'fire', 'global']; // only global if it's called in
+                                              // the runtime context
```
Comment (GA-MEB) [inline lib/diagnostic.js:31]: There isn't any closure here -- `fire` isn't being passed out of the `deathStar` function -- so `laserQuantity` is not accessible anywhere outside of the `deathStar` function scope. However, `fire`'s function scope is inside `deathStar`'s function scope, so something in the `fire` scope can also access `laserQuantity`.

File: lib/diagnostic.js [hunk 2]
```diff
@@ -46,10 +47,19 @@ let scopes = [/* your response here */];
 // method and temporarily bind it (or, change `this`) to `alderaan` before
 // executing it.
 //
+
+let result = deathStar.fire.call(alderaan);
 let fireOnAlderaan = function () {
-  // your code here
+  return result; // couldn't get it to run corectly inside of the function:(
```
Comment (GA-MEB) [inline lib/diagnostic.js:53]: I think you may have mixed up `call` and `bind`. `call` and `apply` both bind the function to a new context AND execute the new function.

You could have written either:

```
let result = deathStar.fire.bind(alderaan);
let fireOnAlderaan = function(){
  return result();
};
```

or just invoked `call`/`apply`.

File: lib/diagnostic.js [hunk 3]
```diff
@@ -46,10 +47,19 @@ let scopes = [/* your response here */];
...
 };
 
+//  Tried this but it didn't work :(
+//
+// let fireOnAlderaan = function () {
+//   deathStar.fire.call(alderaan);
```
Comment (GA-MEB) [inline lib/diagnostic.js:59]: This _does_ work! You probably aren't passing the tests because you didn't return a value from `fireOnAlderaan`.

**Comment by GA-MEB** [comment]
2 / 3 (-0.5 on Q2, Q3)

**Pull request closed without merging**